src/portal/lib/src/config/vulnerability/vulnerability-config.component.ts

```diff
@@ -1,64 +1,93 @@
-import { Configuration } from '../config';
-import {
-    ScanningResultService,
-    SystemInfo,
-    SystemInfoService,
-    ConfigurationService
-} from '../../service/index';
+import { Component, ViewChild, OnInit, OnDestroy } from '@angular/core';
+import { finalize } from "rxjs/operators";
+import { ScanningMetrics } from '../config';
 import { ErrorHandler } from '../../error-handler/index';
-import { isEmptyObject, clone} from '../../utils';
 import { TranslateService } from '@ngx-translate/core';
-import { ClairDetail } from '../../service/interface';
 import { ScanAllRepoService } from './scanAll.service';
 import { OriginCron } from '../../service/interface';
 import { CronScheduleComponent } from "../../cron-schedule/cron-schedule.component";
-const ONE_HOUR_SECONDS: number = 3600;
-const ONE_DAY_SECONDS: number = 24 * ONE_HOUR_SECONDS;
+import { VULNERABILITY_SCAN_STATUS } from "../../utils";
+import { errorHandler as errorHandFn} from "../../shared/shared.utils";
+
 const SCHEDULE_TYPE_NONE = "None";
 @Component({
     selector: 'vulnerability-config',
     templateUrl: './vulnerability-config.component.html',
     styleUrls: ['./vulnerability-config.component.scss', '../registry-config.component.scss']
 })
-export class VulnerabilityConfigComponent implements OnInit {
+export class VulnerabilityConfigComponent implements OnInit, OnDestroy {
     onGoing: boolean;
-    _localTime: Date = new Date();
     originCron: OriginCron;
     schedule: any;
     onSubmitting: boolean = false;
-    config: Configuration;
     openState: boolean = false;
     getLabelCurrent: string;
 
     @ViewChild(CronScheduleComponent, {static: false})
     CronScheduleComponent: CronScheduleComponent;
-
-    @Input()
-
-    @Input() showSubTitle: boolean = false;
-    @Input() showScanningNamespaces: boolean = false;
-    @Output() loadingStatus = new EventEmitter<boolean>();
-    systemInfo: SystemInfo;
-
+    gettingMetrics: boolean;
+    private _scanningMetrics: ScanningMetrics;
+    totalWidth: number = 200;
+    i18nKeyMap = {
+       "Pending": "CONFIG.SCANNING.STATUS.PENDING",
+       "Running": "CONFIG.SCANNING.STATUS.RUNNING",
+       "Stopped": "CONFIG.SCANNING.STATUS.STOPPED",
+       "Error": "CONFIG.SCANNING.STATUS.ERROR",
+       "Success": "CONFIG.SCANNING.STATUS.SUCCESS",
+       "Scheduled": "CONFIG.SCANNING.STATUS.SCHEDULED"
+    };
+    private _loopScheduleInterval;
+    private _loopManualInterval;
     constructor(
-        // private scanningService: ScanningResultService,
         private scanningService: ScanAllRepoService,
         private errorHandler: ErrorHandler,
         private translate: TranslateService,
-        private systemInfoService: SystemInfoService,
-        private configService: ConfigurationService
     ) { }
 
+    get scanningMetrics(): ScanningMetrics {
+        return this._scanningMetrics;
+    }
+    set scanningMetrics(metrics: ScanningMetrics) {
```

Comment:
This scanningMetrics method should be tested. Lots of branches here.

Reply:
More cases have been added.